apps/sms/js/thread_ui.js

```diff
@@ -598,10 +598,11 @@ var ThreadUI = {
       });
     }
 
-    ThreadListUI.mark(threadId, 'read');
-
-    // nothing urgent, let's do it when the main thread has some time
-    setTimeout(MessageManager.markThreadRead.bind(MessageManager, threadId));
+    // Let's mark thread only when thread list is fully rendered and target node
+    // is in the DOM tree.
+    ThreadListUI.whenReady().then(function() {
```

Comment:
For me it looks reasonable to reuse ```ThreadListUI.markReadUnread``` here that does all the job (updates in UI and in DB). And in the most cases we'll have thread list already loaded so it should not have much delay...

But what do you think?